### PR TITLE
Fix for SVN Conflict Changes Deleting Businessflow

### PR DESCRIPTION
**Pull request by aditijagtap**

### CODE REVIEW

File: Ginger/GingerCore/SourceControl/SVNSourceControl.cs [hunk 1]
```diff
@@ -501,15 +506,11 @@ public override bool ResolveConflicts(string Path, eResolveConflictsSide side, r
             if (client == null) Init();
             try
             {
-                if (System.IO.Path.GetExtension(Path) != string.Empty && !System.IO.File.Exists(System.IO.Path.GetFullPath(Path.Replace(".xml", ".ignore"))))
-                {
-                    System.IO.File.Copy(Path, (System.IO.Path.GetFullPath(Path.Replace(".xml", ".ignore"))));
-                }
-                CleanUp(Path);
+                CleanUp(System.IO.Path.GetDirectoryName(Path));
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/SourceControl/SVNSourceControl.cs]: check if cleanup for specific path works

File: Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs [hunk 2]
```diff
@@ -481,6 +483,15 @@ private void FileWatcher_Changed(object sender, FileSystemEventArgs e)
             //Reporter.ToLog(eLogLevel.DEBUG, "FileWatcher change handled: " + e.FullPath + " , " + e.ChangeType);
         }
 
+        public override void ReloadUpdatedXML(string path)
+        {
+            RepositoryItemBase item = null;
+            WaitforFileIsReadable(path);
+            item = GetItemFromCacheByFileName(path);
+            NewRepositorySerializer.ReloadObjectFromFile(item);
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs]: Add null check for item

File: Ginger/GingerCore/SourceControl/SVNSourceControl.cs (hunk 1 above)
Comment (aditijagtap) [reply Ginger/GingerCore/SourceControl/SVNSourceControl.cs]: For SVN we have to give working directory path to Cleanup function. If we pass file path getting "Not a working directory" exception.

File: Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs [hunk 3]
```diff
@@ -481,6 +483,14 @@ private void FileWatcher_Changed(object sender, FileSystemEventArgs e)
             //Reporter.ToLog(eLogLevel.DEBUG, "FileWatcher change handled: " + e.FullPath + " , " + e.ChangeType);
         }
 
+        public override void ReloadUpdatedXML(string path)
+        {
+            WaitforFileIsReadable(path);
+            RepositoryItemBase item = GetItemFromCacheByFileName(path);
+            NewRepositorySerializer.ReloadObjectFromFile(item);
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs]: still missing null check for item

File: Ginger/Ginger/SourceControl/CheckInPage.xaml.cs [hunk 4]
```diff
@@ -240,8 +240,11 @@ private async void CheckInButton_Click(object sender, RoutedEventArgs e)
                                 }
                                 else if (!fi.Locked)
                                 {
+                                    RepositoryFolderBase repositoryFolderBase = WorkSpace.Instance.SolutionRepository.GetRepositoryFolderByPath(Path.GetDirectoryName(fi.Path));
```
Comment (JinendraGhodke) [inline Ginger/Ginger/SourceControl/CheckInPage.xaml.cs]: Need to make it generic as this needs to be called also in other places where update file is needed.
Also test with Git

File: Ginger/GingerCore/SourceControl/SVNSourceControl.cs [hunk 5]
```diff
@@ -258,10 +259,14 @@ public override bool GetLatest(string path, ref string error, ref List<string> c
         {
             if (client == null) Init();
             SvnUpdateResult result;
+
+            RepositoryFolderBase repositoryFolderBase = WorkSpace.Instance.SolutionRepository.GetRepositoryFolderByPath(Path.GetDirectoryName(path));
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/SourceControl/SVNSourceControl.cs]: move this inside try block

File: Ginger/GingerCore/SourceControl/SVNSourceControl.cs [hunk 6]
```diff
@@ -258,10 +259,14 @@ public override bool GetLatest(string path, ref string error, ref List<string> c
...
             try
             {
                 mConflictsPaths.Clear();
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/SourceControl/SVNSourceControl.cs]: move this code inside a function and reuse newly added function wherever you need to call update file

File: Ginger/GingerCoreNET/SourceControl/SourceControlIntegration.cs [hunk 7]
```diff
@@ -174,13 +175,16 @@ public static string GetLockOwner(SourceControlBase SourceControl, string path)
         public static bool ResolveConflicts(SourceControlBase SourceControl, string path, eResolveConflictsSide side)
         {
             string error = string.Empty;
+            RepositoryFolderBase repositoryFolderBase = WorkSpace.Instance.SolutionRepository.GetRepositoryFolderByPath(Path.GetDirectoryName(path));
 
+            repositoryFolderBase.PauseFileWatcher();
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNET/SourceControl/SourceControlIntegration.cs]: Also make a function for client.resolve and call it from here

File: Ginger/GingerCoreNET/SourceControl/SourceControlIntegration.cs [hunk 8]
```diff
@@ -67,12 +68,14 @@ public static bool DeleteFile(SourceControlBase SourceControl, string path)
         public static bool UpdateFile(SourceControlBase SourceControl, string path)
         {
             string error = string.Empty;
-
+            RepositoryFolderBase repositoryFolderBase = WorkSpace.Instance.SolutionRepository.GetRepositoryFolderByPath(Path.GetDirectoryName(path));
+            repositoryFolderBase.PauseFileWatcher();
             if (!SourceControl.UpdateFile(path, ref error))
             {
                 Reporter.ToUser(eUserMsgKey.GeneralErrorOccured, error);
                 return false;
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNET/SourceControl/SourceControlIntegration.cs]: if update file will fail code will return false. And file watcher will remain paused forever

**Pull request merged**